client/components/site-preview-link/index.tsx

```diff
@@ -63,13 +63,17 @@ export default function SitePreviewLink( {
 
 	const { data: previewLinks, isLoading: isFirstLoading } = useSitePreviewLinks( {
 		siteId,
-		onSuccess: ( linksResponse ) => {
-			const validLinksLength = linksResponse?.filter( ( link ) => ! link.isRemoving ).length || 0;
-			setChecked( validLinksLength > 0 );
-		},
 		isEnabled: true,
 	} );
 
+	// Update toggle status if previewLinks have change in another tab
+	useEffect( () => {
+		const validLinksLength = previewLinks?.filter( ( link ) => ! link.isRemoving ).length || 0;
```

Comment:
Nit: if we're only interested in the number of links that are not being removed, we might as well add that number as a dependency of the `useEffect()` and not the entire `previewLinks` object.

Reply:
Thank you! I updated the dependencies array to use the boolean value instead of the length number. Thanks for the feedback ! https://github.com/Automattic/wp-calypso/pull/84658/commits/c369617d6eb0662c1b96d097f41e283a3ac0083f